indexer/base/go.mod

```diff
@@ -0,0 +1,6 @@
+module cosmossdk.io/indexer/base
```

Comment:
Should we maybe just call this `cosmossdk.io/indexer`? I'm not sure what package would be simply `cosmossdk.io/indexer` if it's not this one, although maybe good to leave as `base` to communicate the intent of a base package clearly.